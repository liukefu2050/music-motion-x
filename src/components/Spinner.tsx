import { memo, CSSProperties } from "react"

const Spinner = memo(({ style }: { style?: CSSProperties }) => {
  return (
    <div
      style={{ ...(style || {}) }}
      className=" w-16 h-16 z-[10000] relative my-0 mx-auto mb-0 "
    >
      <div className="bounce w-full h-full rounded-full bg-secondary opacity-60 absolute top-0 left-0 animate-spin-loading" />
      <div
        className="bounce  w-full h-full rounded-full bg-secondary opacity-60 absolute top-0 left-0 animate-spin-loading"
        style={{ animationDelay: "-1s" }}
      />
    </div>
  )
})

Spinner.displayName = "Spinner"

export default Spinner
