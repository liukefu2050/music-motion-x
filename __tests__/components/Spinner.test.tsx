import { render } from "@testing-library/react"
import Spinner, { SpinnerLoading } from "../../src/components/Spinner"

describe("Spinner & SpinnerLoading", () => {
  it("spinner display well", () => {
    const { asFragment } = render(<Spinner style={{ color: "red" }} />)

    expect(asFragment()).toMatchSnapshot()

    const { asFragment: asFragmentOther } = render(<Spinner />)

    expect(asFragmentOther()).toMatchSnapshot()
  })

  it("SpinnerLoading", () => {
    const { asFragment } = render(
      <SpinnerLoading style={{ color: "red" }} color="red" />,
    )

    expect(asFragment()).toMatchSnapshot()

    const { asFragment: asFragmentOther } = render(
      <SpinnerLoading color="red" />,
    )
    expect(asFragmentOther()).toMatchSnapshot()
  })
})
