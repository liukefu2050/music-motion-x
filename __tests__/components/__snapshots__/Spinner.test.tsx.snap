// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner & SpinnerLoading SpinnerLoading 1`] = `
<DocumentFragment>
  <div
    class=" inline-block relative"
    style="color: red;"
  >
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent; animation-delay: -0.45s;"
    />
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent; animation-delay: -0.3s;"
    />
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent; animation-delay: -0.15s;"
    />
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent;"
    />
  </div>
</DocumentFragment>
`;

exports[`Spinner & SpinnerLoading SpinnerLoading 2`] = `
<DocumentFragment>
  <div
    class=" inline-block relative"
  >
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent; animation-delay: -0.45s;"
    />
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent; animation-delay: -0.3s;"
    />
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent; animation-delay: -0.15s;"
    />
    <div
      class=" box-border block absolute w-full h-full rounded-[50%]  animate-ring-spin-loading "
      style="border: 3px solid red; border-color: red transparent transparent transparent;"
    />
  </div>
</DocumentFragment>
`;

exports[`Spinner & SpinnerLoading spinner display well 1`] = `
<DocumentFragment>
  <div
    class=" w-16 h-16 z-[10000] relative my-0 mx-auto mb-0 "
    style="color: red;"
  >
    <div
      class="bounce w-full h-full rounded-full bg-secondary opacity-60 absolute top-0 left-0 animate-spin-loading"
    />
    <div
      class="bounce  w-full h-full rounded-full bg-secondary opacity-60 absolute top-0 left-0 animate-spin-loading"
      style="animation-delay: -1s;"
    />
  </div>
</DocumentFragment>
`;

exports[`Spinner & SpinnerLoading spinner display well 2`] = `
<DocumentFragment>
  <div
    class=" w-16 h-16 z-[10000] relative my-0 mx-auto mb-0 "
  >
    <div
      class="bounce w-full h-full rounded-full bg-secondary opacity-60 absolute top-0 left-0 animate-spin-loading"
    />
    <div
      class="bounce  w-full h-full rounded-full bg-secondary opacity-60 absolute top-0 left-0 animate-spin-loading"
      style="animation-delay: -1s;"
    />
  </div>
</DocumentFragment>
`;
